packages/core/src/components/tag-input/tagInput.tsx

```diff
@@ -262,8 +263,8 @@ export class TagInput extends AbstractPureComponent<ITagInputProps, ITagInputSta
     private addTags = (value: string) => {
         const { inputValue, onAdd, onChange, values } = this.props;
         const newValues = this.getValues(value);
-        let shouldClearInput = inputValue === undefined;
-        shouldClearInput = Utils.safeInvoke(onAdd, newValues) !== false && shouldClearInput;
+        let shouldClearInput =
+            Utils.safeInvoke(onAdd, newValues) !== false && inputValue === TagInput.defaultProps.inputValue;
```

Comment:
ok on second thought, this doesn't work for strings:
```js
{} === {}
> false

[] === []
> false

"" === ""
> true // uh oh - controlled empty value looks like uncontrolled
```

i think we do have to remove the `defaultProps` entry and compare against undefined like you had before